customreboot/systemdBoot.js

```diff
@@ -0,0 +1,122 @@
+import Gio from "gi://Gio";
+import { ExecCommand, Log, LogWarning } from './utils.js';
+
+/**
+ * Represents Systemdboot
+ */
+export class SystemdBoot {
+  /**
+   * Get's all available boot options
+   * @returns {[Map, string]} Map(title, id), defaultOption
+   */
+  static async GetBootOptions() {
+    let bootctl = await this.GetBinary();
+    if (bootctl == "") {
+        Log(`Failed to find bootctl binary`);
+        return undefined;
+    }
+
+    try {
+        let [status, stdout, stderr] = await ExecCommand([bootctl, "list"]);
+        if (status !== 0)
+            throw new Error(`Failed to get list from bootctl: ${status}\n${stdout}\n${stderr}`);
+        Log(`bootctl list: ${status}\n${stdout}\n${stderr}`);
+        let lines = String(stdout).split('\n');
+        let titleRx = /(?<=title:\s+).+/;
+        let idRx = /(?<=id:\s+).+/;
+        let defaultRx = /\(default\)/;
+        let titles = [];
+        let ids = []
+        let defaultOpt;
+        lines.forEach(l => {
+            let title = titleRx.exec(l);
+            let id = idRx.exec(l);
+            if (title && title.length) {
+                titles.push(String(title));
+            } else if (id && id.length) {
+                ids.push(String(id));
+            }
+        });
+        if (titles.length !== ids.length)
+            throw new Error("Number of titles and ids do not match!");
+        let bootOptions = new Map();
+        for (let i = 0; i < titles.length; i++) {
+            bootOptions.set(ids[i], titles[i])
+        }
+
+        bootOptions.forEach((title, id) => {
+            Log(`${id} = ${title}`);
+
+            let defaultRes = defaultRx.exec(title);
+
+            if (defaultRes) {
+                defaultOpt = id;
+            }
+        })
+
+        return [bootOptions, bootOptions.get(defaultOpt)];
+    } catch (e) {
+        LogWarning(e);
+        return undefined;
+    }
+  }
+
+  /**
+   * Set's the next boot option
+   * @param {string} id 
+   * @returns True if the boot option was set, otherwise false
+   */
+  static async SetBootOption(id) {
+    try {
+      let [status, stdout, stderr] = await ExecCommand(
+          ['/usr/bin/pkexec', '/usr/sbin/grub-reboot', id],
```

Comment:
`grub-reboot`? It doesn't seem to be correct here for systemd boot.

Reply:
This is also an issue reflected in the 'upstream', thanks for reminding me. I don't use systemd-boot so haven't tested this yet.